cmd/policy-main.go

```diff
@@ -253,12 +257,62 @@ func accessPermToString(perm accessPerms) string {
 	return policy
 }
 
+func askForConfirmation(msg string) bool {
+	reader := bufio.NewReader(os.Stdin)
+	for {
+		fmt.Printf("%s [y/n]: ", msg)
+		response, err := reader.ReadString('\n')
+		if err != nil {
+			log.Fatal(err)
+		}
+		response = strings.ToLower(strings.TrimSpace(response))
+		if response == "y" || response == "yes" {
+			return true
+		} else if response == "n" || response == "no" {
+			return false
+		}
+	}
+}
+
+func isPermHigher(perm1 string, perm2 string) (ret bool) {
+	switch perm1 {
+	case "none":
+		return false
+	case "readonly", "writeonly":
+		if perm2 == "readwrite" {
+			return false
+		}
+	case "readwrite":
+		if perm2 == "readwrite" {
+			return false
+		}
+	}
+	return true
+}
+
 // doSetAccess do set access.
 func doSetAccess(ctx context.Context, targetURL string, targetPERMS accessPerms) *probe.Error {
 	clnt, err := newClient(targetURL)
 	if err != nil {
 		return err.Trace(targetURL)
 	}
+
+	// Check if there is some overlapping existing rules
+	if !globalJSON {
+		rules, err := clnt.GetAccessRules(ctx)
+		if err != nil {
+			return err.Trace(targetURL)
+		}
+		path := strings.TrimPrefix(clnt.GetURL().Path, "/")
+		for k, v := range rules {
+			if strings.HasPrefix(k, path) && isPermHigher(accessPermToString(targetPERMS), v) {
+				if !askForConfirmation(fmt.Sprintf("You are exposing another sub-resource (`%s`, `%s`), do you confirm? ", k, v)) {
```

Comment:
There should be an option to specify the answer from the commandline `--yes` `--no` or something like that for non-interactive mode.

Reply:
> There should be an option to specify the answer from the commandline `--yes` `--no` or something like that for non-interactive mode.

yes, it makes sense